src/arrayMethodSort.js

```diff
@@ -4,8 +4,43 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    const defaultCompare = (a, b) => {
+      const strA = String(a);
+      const strB = String(b);
+
+      if (strA < strB) {
+        return -1;
+      }
+
+      if (strA > strB) {
+        return 1;
+      }
+
+      return 0;
+    };
+
+    const compare = compareFunction || defaultCompare;
+
+    const array = this.slice();
+    const len = array.length;
+
+    for (let i = 0; i < len - 1; i++) {
+      for (let j = 0; j < len - 1 - i; j++) {
+        if (compare(array[j], array[j + 1]) > 0) {
+          const temp = array[j];
+
+          array[j] = array[j + 1];
+          array[j + 1] = temp;
+        }
+      }
```

Comment:
The implementation uses the bubble sort algorithm, which is not efficient for large datasets due to its O(n^2) complexity. For better performance on larger arrays, consider implementing a more efficient sorting algorithm like quicksort or mergesort.